src/main/java/org/launchcode/techjobs/persistent/controllers/HomeController.java

```diff
@@ -36,9 +54,17 @@ public String processAddJobForm(@ModelAttribute @Valid Job newJob,
 
         if (errors.hasErrors()) {
             model.addAttribute("title", "Add Job");
+            model.addAttribute("employers", employerRepository.findAll());
             return "add";
         }
 
+        Optional<Employer> employer = employerRepository.findById(employerId);
+        if( employer.isPresent()){
+            Employer e = employer.get();
+            newJob.setEmployer(e);
+        }
+        List<Skill> skillObjs = (List<Skill>) skillRepository.findAllById(skills);
+        newJob.setSkills(skillObjs);
```

Comment:
We've updated the newJob object and set the employer and skill fields, but there is still a specific method that must be called in order to save it to the database. HINT: look at employerConroller > processAddEmployerForm method